src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -442,9 +445,26 @@ private void CancelRequestAbortedToken()
         {
             try
             {
-                _abortedCts.Cancel();
-                _abortedCts.Dispose();
-                _abortedCts = null;
+                CancellationTokenSource localAbortCts = null;
+                lock (_abortLock)
+                {
+                    if (_abortedCts != null)
+                    {
+                        if (!_preventRequestAbortedCancellation)
```

Comment:
In the case when _preventRequestAbortedCancellation is true here, we should dispose _abortedCts (outside the lock ofc) in addition to nulling it out.